src/boot/efi/pe.c

```diff
@@ -153,15 +155,32 @@ static void locate_sections(
         assert(offsets);
         assert(sizes);
 
+        EFI_STATUS err;
+
         for (size_t i = 0; i < n_table; i++) {
                 const PeSectionHeader *sect = section_table + i;
 
                 for (size_t j = 0; sections[j]; j++) {
                         if (memcmp(sect->Name, sections[j], strlen8(sections[j])) != 0)
                                 continue;
 
-                        offsets[j] = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
-                        sizes[j] = sect->VirtualSize;
+                        size_t offset = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
+                        size_t size = sect->VirtualSize;
+
+                        offsets[j] = offsets[j] ? : offset;
+                        sizes[j] = sizes[j] ? : size;
+
+                        if (in_memory && streq8(sections[j], ".dtb")) {
+                                err = devicetree_match(base + offset, size);
+
```

Comment:
in systemd we have this coding style rule that function calls and their immediate error handling should be kept together, i.e. empty lines may be inserter before the func call, and after the error handling, but not between the call and the error handline. the error checking below is separated from the function call here by an empty line though, please remove